cachelib/navy/common/FdpNvme.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <folly/File.h>
+#include <folly/experimental/io/AsyncBase.h>
+#include <folly/experimental/io/IoUring.h>
+
+#include "cachelib/navy/common/Buffer.h"
+#include "cachelib/navy/common/Device.h"
+
+#ifndef CACHELIB_IOURING_DISABLE
+#include <liburing.h>
+#endif
```

Comment:
FYI. Actually, we should not include `liburing.h` directly here due to some internal dependency issues causing test failures on some targets. I will remove this.

Reply:
Ok. 
Then should we include <liburing.h> through <folly/experimental/io/IoUring.h> which include <liburing.h> in the file?

Reply:
@arungeorge83 Yeah, right. It will include the `liburing.h` from the repo instead of that installed on the system.

FYI, I am taking care of minor changes including those required for the internal build, so you don't need to update the commit.